### PR TITLE
Gh pages first angular page

### PR DESCRIPTION
**Pull request by RomE-HK**

### CODE REVIEW

**Comment by RomE-HK** [comment]
Test variant to verify that it could work this way

File: angular/app.component.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            new Item("Хлеб",15.9),
+            new Item("Масло",60),
+            new Item("Картофель",22.6, true),
+            new Item("Сыр",310)
+        ];
+    }
+});
+ 
+AppComponent.prototype.addItem = function (text, price) {
+        if (text == undefined || text.trim() == "" || price == undefined)
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Unexpected use of undefined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 17
           

Complexity increasing per file
==============================
- angular/app.component.js  4
- angular/app.module.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370&bid=17170180)

File: angular/app.module.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+var AppModule = ng.core.NgModule({
+    imports: [ng.platformBrowser.BrowserModule, ng.forms.FormsModule],
+    declarations: [AppComponent],
```
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['AppComponent' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+var AppModule = ng.core.NgModule({
+    imports: [ng.platformBrowser.BrowserModule, ng.forms.FormsModule],
```
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
+function Item(purchase, price, done) {
+    this.purchase = purchase;
+    this.price = price;
+    this.done = done;
+}
+ 
+var AppComponent = ng.core.Component({
+    selector: 'my-app',
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js (hunk 1 above)
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                <tr *ngFor="let item of items">
+                    <td>{{item.purchase}}</td>
+                    <td>{{item.price}}</td>
+                    <td><input type="checkbox" [(ngModel)]="item.done" /></td>
+                </tr>
+            </tbody>
+        </table>
+    </div>`
+    })
+  .Class({
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+var AppModule = ng.core.NgModule({
```
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    constructor: function() {
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Expected method shorthand.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/main.js [hunk 8]
```diff
@@ -0,0 +1,2 @@
+ng.platformBrowserDynamic.platformBrowserDynamic()
+    .bootstrapModule(AppModule);
```
Comment (codacy-review) [inline angular/main.js]: Codacy Issue found: ['AppModule' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
+function Item(purchase, price, done) {
+    this.purchase = purchase;
+    this.price = price;
+    this.done = done;
+}
+ 
+var AppComponent = ng.core.Component({
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+    imports: [ng.platformBrowser.BrowserModule, ng.forms.FormsModule],
+    declarations: [AppComponent],
+    bootstrap: [AppComponent]
+  })
+  .Class({
+    constructor: function() { }
```
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: [Expected method shorthand.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/main.js [hunk 11]
```diff
@@ -0,0 +1,2 @@
+ng.platformBrowserDynamic.platformBrowserDynamic()
```
Comment (codacy-review) [inline angular/main.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 6 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
+var AppModule = ng.core.NgModule({
+    imports: [ng.platformBrowser.BrowserModule, ng.forms.FormsModule],
+    declarations: [AppComponent],
+    bootstrap: [AppComponent]
+  })
+  .Class({
```
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+var AppModule = ng.core.NgModule({
+    imports: [ng.platformBrowser.BrowserModule, ng.forms.FormsModule],
+    declarations: [AppComponent],
+    bootstrap: [AppComponent]
```
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['AppComponent' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js (hunk 1 above)
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Expected { after 'if' condition.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 17
           

Complexity increasing per file
==============================
- angular/app.component.js  4
- angular/app.module.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370&bid=17170180)

File: angular/app.module.js (hunk 13 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['AppComponent' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/main.js (hunk 8 above)
Comment (codacy-review) [inline angular/main.js]: Codacy Issue found: ['AppModule' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js (hunk 9 above)
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 3 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/main.js (hunk 11 above)
Comment (codacy-review) [inline angular/main.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+            new Item("Bread",15.9),
+            new Item("Butter",60),
+            new Item("Potatoes",22.6, true),
+            new Item("Cheese",310)
+        ];
+    }
+});
+ 
+AppComponent.prototype.addItem = function (text, price) {
+        if (text == undefined || text.trim() == "" || price == undefined)
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Expected { after 'if' condition.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 10 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: [Expected method shorthand.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 6 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['ng' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 12 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+                    <td>{{item.purchase}}</td>
+                    <td>{{item.price}}</td>
+                    <td><input type="checkbox" [(ngModel)]="item.done" /></td>
+                </tr>
+            </tbody>
+        </table>
+    </div>`
+    })
+  .Class({
+    constructor: function() {
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Expected method shorthand.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js (hunk 9 above)
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js (hunk 4 above)
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+                <tr *ngFor="let item of items">
+                    <td>{{item.purchase}}</td>
+                    <td>{{item.price}}</td>
+                    <td><input type="checkbox" [(ngModel)]="item.done" /></td>
+                </tr>
+            </tbody>
+        </table>
+    </div>`
+    })
+  .Class({
```
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 6 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.component.js (hunk 14 above)
Comment (codacy-review) [inline angular/app.component.js]: Codacy Issue found: [Unexpected use of undefined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

File: angular/app.module.js (hunk 2 above)
Comment (codacy-review) [inline angular/app.module.js]: Codacy Issue found: ['AppComponent' is not defined.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5238370)

**Pull request merged**